lua/lib/filters.lua

```diff
@@ -0,0 +1,158 @@
+------------
+--- @module filters
+--
+-- some filters that could be useful for musical event processing
+
+local f = {}
+f.__index = f
+
+-- helper to increment and wrap an index
+local function wrap_inc(x, max)
+   local y = x + 1
+   while y > max do y = y - max end
+   return y
+end
+
+--- clear a filter's history
+function f:clear() 
+   for i=1,self.bufsize do self.buf[i]=0 end
+end
+
+-- debug function to print the buffer
+function f:print_string(pre)
```

Comment:
For what it is worth if we do find value in printable representations of lua objects it might be worth consistently using implementing the `__tostring` meta method:
https://www.lua.org/pil/13.3.html

I personally see value in implementing `__tostring` in may places to make it easier to interactively debug stuff.